pkg/controller/cloud/node_controller.go

```diff
@@ -438,21 +438,24 @@ func excludeTaintFromList(taints []v1.Taint, toExclude v1.Taint) []v1.Taint {
 	return newTaints
 }
 
-// ensureNodeExistsByProviderIDOrInstanceID first checks if the instance exists by the provider id and then by calling instance id with node name
+// ensureNodeExistsByProviderIDOrInstanceID checks if the instance exists by the provider id,
+// If provider id is empty it calls instanceId with node name to get provider id
 func ensureNodeExistsByProviderIDOrInstanceID(instances cloudprovider.Instances, node *v1.Node) (bool, error) {
```

Comment:
Also rename the function to `ensureNodeExistsByProviderID()`?

Reply:
Yeah, the name is indeed confusing now. It still indicates the ensure operation involves both providerID and instanceID.

Do we suppose 'instanceID' and 'providerID' to be the same thing? If it is I think it's fine to just name 'ByProviderID'

Currently it seems the case is 'InstanceID' returned by cloud provider is used as 'ProviderID'.
Related:
https://github.com/kubernetes/kubernetes/issues/60466